x/programs/wasmlanche/src/mock.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+#[cfg(feature = "test")]
+mod functions {
+    use crate::context::CallContractArgs;
+    use crate::{types::Gas, Id};
+    use crate::{Address, Context, ExternalCallError};
+    use borsh::BorshSerialize;
+
+    impl Context {
+        /// Mocks an external function call.
+        /// # Panics
+        /// Panics if serialization fails.
+        pub fn mock_call_function<T, U>(
+            &self,
+            target: Address,
+            function: &str,
+            args: T,
+            max_units: Gas,
+            max_value: u64,
+            result: U,
+        ) where
+            T: BorshSerialize,
+            U: BorshSerialize,
```

Comment:
Should there be a `BorshDeserialize` bound somewhere here too?

Reply:
nothing is getting deserialized in the function. could add to make it clear that it will eventually get deserialized?

Reply:
Fine for now, I think somewhere there's supposed to be one. We don't want to serialize then not be able to deserialize.